sentry-android-core/src/main/java/io/sentry/android/core/EnvelopeFileObserver.java

```diff
@@ -0,0 +1,42 @@
+package io.sentry.android.core;
+
+import android.os.FileObserver;
+import io.sentry.core.*;
+import io.sentry.core.util.Objects;
+
+final class EnvelopeFileObserver extends FileObserver {
+
+  private String rootPath;
+  private final IEnvelopeSender envelopeSender;
+  private ILogger logger;
+
+  // The preferred overload (Taking File instead of String) is only available from API 29
+  @SuppressWarnings("deprecation")
+  EnvelopeFileObserver(
+      String path,
+      IEnvelopeSender envelopeSender,
+      ILogger logger) {
+    super(path);
+    this.rootPath = Objects.requireNonNull(path, "File path is required.");
+    this.envelopeSender = Objects.requireNonNull(envelopeSender, "Envelope sender is required.");
+    this.logger = Objects.requireNonNull(logger, "Logger is required.");
+  }
+
+  @Override
+  public void onEvent(int eventType, @android.support.annotation.Nullable String relativePath) {
+    if (relativePath == null) {
+      return;
+    }
+
+    logger.log(
+      SentryLevel.DEBUG,
+      "onEvent fired for EnvelopeFileObserver with event type %d on path: %s for file %s.",
+      eventType,
+      this.rootPath,
+      relativePath);
+
+    // TODO: Only some event types should be pass through?
```

Comment:
For sure need to deal with the right events, they are:

> Value is either 0 or a combination of ACCESS, MODIFY, ATTRIB, CLOSE_WRITE, CLOSE_NOWRITE, OPEN, MOVED_FROM, MOVED_TO, CREATE, DELETE, DELETE_SELF, and MOVE_SELF

https://developer.android.com/reference/android/os/FileObserver.html

I guess since I delete them on each run, it'll raise another event immediately for that delete.

Reply:
if I am not wrong, only `CREATE` right, and when you cache to disk (an event) and it's already enqueued, you still get an event from the Observer, we will need to take care of it to not duplicate the events in the queue

Reply:
The file is deleted once we handle it so unless it's fired in parallel, it won't end up as duplicate events. But in regards it would be hitting the `FileNotFound` so we should opt-in to some event types. IIRC per spec it would be a file rename where it's not prefixed by `.`